### PR TITLE
"Concepts" edit suggestions

### PR DESCRIPTION
**Pull request by lann**
(also a Intro typo fix)

### CODE REVIEW

File: component-model/src/concepts.md [hunk 1]
```diff
@@ -21,25 +21,25 @@ A **world** is a higher-level contract that describes a component's capabilities
...
 * A "regex parser" world would export a "parse regex" function, and would import nothing. This declares not only that the component implementing this world can parse regular expressions, but also that it calls no other APIs. A user of such a parser could know, without looking at the implementation, that is does not access the file system, or send the user's regexes to a network service.
 
 ## Components
 
-Physically, a **component** is a specially formatted WebAssembly file. Internally, the component could include multiple traditional ("core") WebAssembly modules, and sub-components, composed via their imports and exports.
+Physically, a **component** is a specially-formatted WebAssembly file. Internally, the component could include multiple traditional ("core") WebAssembly modules, and sub-components, composed via their imports and exports.
 
 The external interface of a component - its imports and exports - corresponds to a world. The component, however, defines behaviour and internal state.
```
Comment (lann) [inline component-model/src/concepts.md:38]: > The component, however, defines behaviour and internal state.

I'm not sure what this sentence is saying...

File: component-model/src/concepts.md [hunk 2]
```diff
@@ -21,25 +21,25 @@ A **world** is a higher-level contract that describes a component's capabilities
...
-* The "WASI (WebAssembly System Interface) command line" world is a classic example of a "hosting environment" use of the world concept. This world exports APIs such as file I/O, sockets, random number generation, and other POSIX-style functionality, so that application components that depend on this world - that is, command line applications - can use those familiar capabilities.
+* A (trivial) "HTTP proxy" world would export a "receive HTTP requests" interface, and import a "send HTTP requests" interface. A host, or another component, would call the exported "receive" interface, passing an HTTP request; the component would forward it on via the imported "send" interface. To be a _useful_ proxy, the component may also need to import interfaces such as I/O and clock time - without those imports the component could not perform, for example, on-disk caching.
```
Comment (lann) [inline component-model/src/concepts.md:30]: A proxy _could_ be useful without any other imports (e.g. gzipping responses) and could even perform in-memory caching (if the instance persisted between requests).

File: component-model/src/concepts.md (hunk 1 above)
Comment (itowlson) [reply component-model/src/concepts.md:38]: The goal was to emphasise that the world defined the interface contract, but the component defined how the contract was implemented.  Whaddaya know, that might be a clearer way of putting it, right there.

File: component-model/src/concepts.md [hunk 3]
```diff
@@ -21,25 +21,25 @@ A **world** is a higher-level contract that describes a component's capabilities
...
+* The "WASI (WebAssembly System Interface) command line" world is a classic example of a "hosting environment" use of the world concept. This world exports APIs such as file I/O, sockets, random number generation, and other POSIX-style functionality, so that application components that depend on this world - such as command line applications - can use those familiar capabilities.
```
Comment (lann) [inline component-model/src/concepts.md:31]: WASI is available to non-command-line applications too, i.e. "reactors".

Comment (lann) [reply component-model/src/concepts.md:31]: Woops, I just re-read this section... :facepalm:

File: component-model/src/concepts.md (hunk 1 above)
Comment (lann) [reply component-model/src/concepts.md:38]: I'll update.

**Review by itowlson — APPROVED** [review]

**Pull request merged**